src/data/nat/factorization.lean

```diff
@@ -217,6 +217,18 @@ begin
   simp [list.eq_of_mem_repeat hq],
 end
 
+lemma pow_factorization_le {n : ℕ} (p : ℕ) (hn : 0 < n) : p ^ n.factorization p ≤ n :=
```

Comment:
Probably ought to use `n ≠ 0` here instead of `0 < n`, and below. The same applies to `le_of_dvd` itself but we can leave that for now if you prefer

Reply:
Oh, of course.  I'd somehow forgotten which way the style guide requires.